api/client/squash.go

```diff
@@ -0,0 +1,50 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/docker/docker/api/types"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/stringid"
+)
+
+// CmdSquash merges filesystem layers of an image into a new image.
+//
+// Usage: docker squash IMAGE [ANCESTOR]
+func (cli *DockerCli) CmdSquash(args ...string) error {
+	cmd := cli.Subcmd("squash", []string{"IMAGE [ANCESTOR]"}, "Merge filesystem layers of an image into a new image", true)
+	cmd.Require(flag.Min, 1)
+	cmd.Require(flag.Max, 2)
+
+	noTrunc := cmd.Bool([]string{"#notrunc", "-no-trunc"}, false, "Don't truncate output")
```

Comment:
oh, is that what the `#notrunc` means? I actually just copied this section from the implementation of the `docker history` command. :disappointed: 

... 

Just got back from reading the comments/docs on `pkg/mflag`. It all makes sense now :dizzy_face: